app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,61 @@
+class PasswordResetsController < ApplicationController
+  before_action :provide_user, only: [:edit, :update]
+  before_action :valid_user?, only: [:edit, :update]
```

Comment:
by naming the method like this, it should return true/false but `valid_user?` method does not return true/false. It does either a redirect, either it returns nil